lib/ui/profitloss/RecordAmountListItem.dart

```diff
@@ -0,0 +1,357 @@
+      SizedBox(width: style.iconLineSpace),
+    ];
+
+    if (viewModel.selectedDate != null) {
+      _buildDatePicker(listBuilder, viewModel, style);
+    }
+
+    if (viewModel.listOfCrops != null) {
+      _buildItemPicker(listBuilder, viewModel, style);
+    }
+
+    if (viewModel.selectedDate == null &&
+        viewModel.listOfCrops == null) {
+      return _buildDescriptionTextField(listBuilder, viewModel, style);
+    }
+
+    if (viewModel.isEditable) {
+      listBuilder.add(SizedBox(width: style.iconLineSpace));
+      listBuilder.add(Row(
+          mainAxisAlignment: MainAxisAlignment.end,
+          children: <Widget>[
+            Image.asset(viewModel.arrow,
+                height: style.detailTextSpacing)
+          ]));
+    }
+
+    return listBuilder;
+  }
+
+  List<Widget> _buildDescriptionTextField(List<Widget> listBuilder, RecordAmountListItemViewModel viewModel, RecordAmountListItemStyle style) {
+    listBuilder.add(Expanded(
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          viewModel.isEditable
+              ? TextField(
+                  decoration: InputDecoration(
+                      hintText: viewModel.hint,
+                      hintStyle: style.pendingDetailTextStyle,
+                      border: InputBorder.none,
+                      contentPadding: style.cardMargins,
+                      counterText: ""),
+                  textAlign: TextAlign.left,
+                  style: style.detailTextStyle,
+                  maxLines: style.maxLines,
+                  controller: _textFieldController,
+                  onEditingComplete: () => _checkTextField(viewModel),
+                  enabled:viewModel.isEditable,
+                )
+              : Text(viewModel.description,
+                  textAlign: TextAlign.left,
+                  style: style.detailTextStyle)
+        ],
+      ),
+    ));
+    return listBuilder;
+  }
+
+  void _buildItemPicker(List<Widget> listBuilder, RecordAmountListItemViewModel viewModel, RecordAmountListItemStyle style) {
+    listBuilder.add(
+        Text(viewModel.title, style: style.titleTextStyle));
+    listBuilder.add(Expanded(
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.end,
+        crossAxisAlignment: CrossAxisAlignment.end,
+        children: <Widget>[
+          viewModel.isEditable
+              ? DropdownButtonHideUnderline(
+                  child: DropdownButton(
+                    value: viewModel.selectedItem,
+                    style: style.detailTextStyle,
+                    items: viewModel.isEditable
+                        ? _getDropDownMenuItems(viewModel)
+                        : null,
+                    onChanged: _changeDropDownItem,
+                    hint: Text(viewModel.hint,
+                        style: style.pendingDetailTextStyle),
+                    icon: Icon(Icons.lens, size: 0),
```

Comment:
size 0?

Reply:
DropdownButton should have an Icon, I've changed it and set it to null.